### PR TITLE
Support dedicated tenancy for ec2 instances

### PR DESCRIPTION
**Pull request by daveseff**
This is to fix issue #259 picking up where it was left off. I have tested this in my environment and seems to works just fine. This should supercede PR #147

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -372,6 +372,9 @@ The name of the key pair associated with this instance. This must be an existing
 #####`instance_type`
 *Required* The type to use for the instance. This parameter is set at creation only; it is not affected by updates. See [Amazon EC2 Instances](http://aws.amazon.com/ec2/instance-types/) for available types.
 
+#####`tenancy`
+*Optional* Dedicated instances are Amazon EC2 instances that run in a virtual private cloud (VPC) on hardware that's dedicated to a single customer. Detaults to shared hardware.
```
Comment (DavidS) [inline README.md]: "Detaults" => "Defaults" ; maybe also mention how the param relates to dedicated instances?

File: spec/acceptance/instance_spec.rb [hunk 2]
```diff
@@ -137,6 +138,7 @@ def get_instance(name)
           :project    => 'cloud',
           :created_by => 'aws-acceptance'
         },
+        :tenancy => 'dedicated',
```
Comment (DavidS) [inline spec/acceptance/instance_spec.rb]: What requirements adds this to the AWS account the tests are running against? Does that mean the test account needs dedicated instances?

File: spec/unit/type/ec2_instance_spec.rb [hunk 3]
```diff
@@ -22,6 +22,7 @@
       :region,
       :availability_zone,
       :monitoring,
+      :tenency,
```
Comment (DavidS) [inline spec/unit/type/ec2_instance_spec.rb]: typo: tenancy

File: spec/unit/type/ec2_instance_spec.rb [hunk 4]
```diff
@@ -145,6 +146,7 @@
     'image_id',
     'availability_zone',
     'instance_type',
+    'tenancy',
```
Comment (DavidS) [inline spec/unit/type/ec2_instance_spec.rb]: This doesn't match what the property actually does. Just remove it.

**Comment by DavidS** [comment]
Please address the inline comments.

Assuming that the placement cannot be changed after the fact, adding tenancy to the list and test at https://github.com/puppetlabs/puppetlabs-aws/blob/7dff5f6d2cded2fcb2418a3969eae0c3f619a813/spec/acceptance/instance_spec.rb#L176-L193 is probably also indicated.

**Comment by daveseff** [comment]
Thanks @DavidS . The tests were written very hastily. I suck at writing tests :)  I needed to get this support in for my company. I'm going back and correcting these now.

**Comment by DavidS** [comment]
Thanks for your work!

**Comment by pjmorr** [comment]
is this supposed to work now?  I receive an error when trying to create an ec2_instance with tenancy => 'dedicated' in the config.

```
Error: no parameter named 'tenancy' at filename.pp:96 on Ec2_instance[serverfqdn] at filename:96 on node mycomputername.net
```

I see the test case create.pp and destroy.pp and have even tried using that to see if it works and still receive the same error.  Any idea what I am doing wrong?

**Pull request merged**